packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -453,21 +467,30 @@ class AndroidSdk {
     if (_sdkManagerEnv == null) {
       // If we can locate Java, then add it to the path used to run the Android SDK manager.
       _sdkManagerEnv = <String, String>{};
-      final String? javaBinary = findJavaBinary(
+      final String? path = findJavaBinary(
         androidStudio: globals.androidStudio,
         fileSystem: globals.fs,
         operatingSystemUtils: globals.os,
         platform: globals.platform,
       );
-      if (javaBinary != null && globals.platform.environment['PATH'] != null) {
-        _sdkManagerEnv!['PATH'] = globals.fs.path.dirname(javaBinary) +
-                                 globals.os.pathVarSeparator +
-                                 globals.platform.environment['PATH']!;
+      if (path != null && globals.platform.environment['PATH'] != null) {
+        _sdkManagerEnv!['PATH'] = globals.fs.path.dirname(path) +
+                                  globals.os.pathVarSeparator +
+                                  globals.platform.environment['PATH']!;
       }
     }
     return _sdkManagerEnv!;
   }
 
+  String? get javaHome {
+    return findJavaHome(
```

Comment:
nit: perhaps put this just above `findJavaHome`. That also means `sdkManagerVersion` remains next to `sdkManagerEnv`.